specs/interop/dependency-set.md

```diff
@@ -47,6 +50,39 @@ It is a known issue that not all software in the Ethereum ecosystem can handle 3
 The dependency set is managed in the client software. Adding a chain to the dependency set is
 considered an upgrade to the network. It is not possible to remove chains from the dependency set.
 
+During an upgrade, only the derivation pipeline (impersonating the `DEPOSITOR_ACCOUNT`)
+can initiate dependency additions on L2.
+
+The dependency set is managed through a two-step process involving both L2 and L1 contracts:
+
+1. The L2 `DependencyManager` predeploy contract initiates the addition of a new dependency through a withdrawal transaction
+2. The L1 `SuperchainConfig` contract processes this withdrawal and updates the L1-side dependency set
+
+## Architecture
+
+### L2 DependencyManager
+
+The L2 `DependencyManager` is a predeploy contract (0x4200000000000000000000000000000000000029)
+that manages the L2-side dependency set. It:
+
+- Maintains the list of chain IDs in the dependency set
```

Comment:
it might be worth noting the chain ID + Superchain config + System Config are all tracked together in unison